etcdserver/etcdserverpb/raft_internal.proto

```diff
@@ -58,6 +59,9 @@ message InternalRaftRequest {
   AuthRoleGetRequest auth_role_get = 1202;
   AuthRoleGrantPermissionRequest auth_role_grant_permission = 1203;
   AuthRoleRevokePermissionRequest auth_role_revoke_permission = 1204;
+
+  membershippb.ClusterVersionSetRequest cluster_version_set = 1300;
+  membershippb.ClusterMemberAttrSetRequest cluster_member_attr_set = 1301;
```

Comment:
these are unknown fields for older etcd? What happens to the older etcd if it receives this request?

Reply:
Ok, now I see. These will be ignored in old etcd, and it instead uses v2 API.

Reply:
Looks like these two fields made v3.5 to v3.4 downgrade fail. 

Can we make it an experimental API in v3.4 and add these two fields to make downgrade from v3.5 to v3.4 possible? 

```
panic: not implemented

goroutine 214 [running]:
go.etcd.io/etcd/etcdserver.(*applierV3backend).Apply(0xc000238468, 0xc0002b2460, 0x0)
        /home/chaochn/workplace/EKS-etcd/src/EKS-etcd/etcdserver/apply.go:173 +0xf85
go.etcd.io/etcd/etcdserver.(*authApplierV3).Apply(0xc000242000, 0xc0002b2460, 0x0)
        /home/chaochn/workplace/EKS-etcd/src/EKS-etcd/etcdserver/apply_auth.go:60 +0xd2
go.etcd.io/etcd/etcdserver.(*EtcdServer).applyEntryNormal(0xc000290600, 0xc0002d14d8)
        /home/chaochn/workplace/EKS-etcd/src/EKS-etcd/etcdserver/server.go:2230 +0x1fb
go.etcd.io/etcd/etcdserver.(*EtcdServer).apply(0xc000290600, 0xc000302a80, 0xa, 0xc, 0xc0001c8000, 0x0, 0xc000046000, 0xc00033f640)
        /home/chaochn/workplace/EKS-etcd/src/EKS-etcd/etcdserver/server.go:2144 +0x579
go.etcd.io/etcd/etcdserver.(*EtcdServer).applyEntries(0xc000290600, 0xc0001c8000, 0xc00002c000)
        /home/chaochn/workplace/EKS-etcd/src/EKS-etcd/etcdserver/server.go:1396 +0xe5
go.etcd.io/etcd/etcdserver.(*EtcdServer).applyAll(0xc000290600, 0xc0001c8000, 0xc00002c000)
        /home/chaochn/workplace/EKS-etcd/src/EKS-etcd/etcdserver/server.go:1120 +0x88
go.etcd.io/etcd/etcdserver.(*EtcdServer).run.func8(0x11b6e30, 0xc0005da040)
        /home/chaochn/workplace/EKS-etcd/src/EKS-etcd/etcdserver/server.go:1065 +0x3c
go.etcd.io/etcd/pkg/schedule.(*fifo).run(0xc0003c0180)
        /home/chaochn/workplace/EKS-etcd/src/EKS-etcd/pkg/schedule/schedule.go:157 +0xf3
created by go.etcd.io/etcd/pkg/schedule.NewFIFOScheduler
        /home/chaochn/workplace/EKS-etcd/src/EKS-etcd/pkg/schedule/schedule.go:70 +0x13b
```

Reply:
Do you know which one failed: 
  - ClusterVersionSetRequest or  ClusterMemberAttrSetRequest

`ClusterMemberAttrSetRequest` is only called by publishV3, that is not yet called in v3.5. so I assume not this one.

`ClusterVersionSetRequest` seems to be called from: https://github.com/etcd-io/etcd/blob/f82b5cb7768dacad9fb310232c1383b4e6718378/server/etcdserver/server.go#L2431

@hexfusion @wenjiaswe @ychen11 shell we postpone enabling `monitorVersions()` till etcd-3.6 ?

Reply:
Yea, I confirmed in v3.4 we haven't yet implemented `ClusterVersionSet` API just now. And the raftRequest MustUnmarshal just slipped this through, which is surprising to me.  

```
case r.ClusterVersionSet != nil:
    a.s.applyV3Internal.ClusterVersionSet(r.ClusterVersionSet)
default:
    panic("not implemented")
```

so do we want to make a equivalent `monitorVersionsV3()` in etcd-3.5? @ptabor

Reply:
> shell we postpone enabling monitorVersions() till etcd-3.6 ?

Yes, this breaks upgrade case, which is not acceptable for 3.5 + 3.5 + 3.4.

Thanks for the catch @chaochn47.

@chaochn47 Can you help making this optional?

Reply:
FTR: https://github.com/etcd-io/etcd/issues/12987